nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/state/reporting-tasks/reporting-tasks.reducer.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createReducer, on } from '@ngrx/store';
+import { ReportingTasksState } from './index';
+import {
+    createReportingTask,
+    createReportingTaskSuccess,
+    deleteReportingTaskSuccess,
+    loadReportingTasks,
+    loadReportingTasksSuccess,
+    reportingTasksApiError,
+    startReportingTaskSuccess,
+    stopReportingTaskSuccess
+} from './reporting-tasks.actions';
+import { produce } from 'immer';
+
+export const initialState: ReportingTasksState = {
+    reportingTasks: [],
+    saving: false,
+    loadedTimestamp: '',
+    error: null,
+    status: 'pending'
+};
```

Comment:
We might want to consider adding a selected property to the state or to the reporting task in the list of the one that is selected. This might be a bigger change as it would imply a refactoring of how selection is handled. At the moment, selection is not persisted and thus makes it challenging to programmatically select an item in the list as a part of another action. Specifically when an item is added to the list as a result of a creation action. The list is updated in the store, but there is no way to indicate it should be selected as a result.

If I'm missing something, let me know and we can discuss.

Reply:
The source of truth for selection is the `router-state`. If there is a case where we need a Reporting Task selected, we need to dispatch a `selectReportingTask` action. It's very possible that we're missing this in a few spots as the Reporting Task listing isn't completed just yet.